probatus/utils/shap_helpers.py

```diff
@@ -200,3 +201,66 @@ def calculate_shap_importance(shap_values, columns, output_columns_suffix=""):
     importance_df = importance_df.sort_values(f"mean_abs_shap_value{output_columns_suffix}", ascending=False)
 
     return importance_df
+
+
+def get_shap_cv_values(model, X_train, y_train, cv, approximate=False, check_additivity=True, **shap_kwargs):
+    """
+    Calculates SHAP values of a model via cross validation.
+
+    Args:
+        model (binary model):
+            Trained model.
+
+        X_train (pd.DataFrame or np.ndarray):
+            Model features.
+
+        y_train (pd.Series or np.ndarray):
+            Model target.
+
+        cv (int or cross-validation generator):
+            If int, use StratifiedKFolds with k = cv. Otherwise, use CV generator (aka CV splitter) passed.
+
+        approximate (bool):
+            If True, uses SHAP approximations - less accurate, but very fast. It applies to tree-based explainers only.
+
+        check_additivity (bool):
+            If False, SHAP will disable the additivity check for tree-based models.
+
+        **shap_kwargs: kwargs of the shap.Explainer
+
+    Returns:
+        shap_values_cv
+    """
+    if isinstance(cv, int):
+        cv = StratifiedKFold(n_splits=cv, shuffle=True, random_state=0)
+
+    shap_values_cv = []
+
+    for train_index, test_index in cv.split(X_train, y_train):
+
+        X_train_cv = X_train.iloc[train_index]
+        y_train_cv = y_train.iloc[train_index]
+
+        model.fit(X_train_cv, y_train_cv)
```

Comment:
Good to check if the model is a search optimization, see ShapRFECV how we dealt with that. 
If it is optimization like RandomizedSearchCV, you have two options:
- Optimize once on the entire dataset, with using CV that was passed by the user. Once you optimize parameters you run each stage of CV with the model with optimized parameters. Thanks to this, at each iteration you end up with model with the same hyperparameters (more stable results over folds), and it is much faster because you don't need to optimize it each time. We chose this option
- Optimize within each CV iteration, which is problematic because you need to do CV inside of CV- which is much slower and less stable over folds.

Handling optimization would be great for this function, because then we can use it inside ShapRFECV and others, otherwise, it will be a standalone function that we don't use anywhere else in probatus.